Fix reduced volume issue for users of PCM5122 DAC

hmbedded opened the pull request:
Since the recent upgrade to the 3.18 kernel users of the PCM5122 DAC have found that there output volume has been significantly reduced. The issue is related to commit id 5be2fc2 which added support for a couple of analogue gain selections. One of these (Playback Volume) sets a gain of either 0db or -6dB and is intended to set the full scale output at 2Vrms or 1Vrms respectively. The reset state is 0db (so 2Vrms) but unfortunately "alsactl restore" applies a default of -20dB to a control named "Playback Volume" if '/var/lib/alsa/asound.state' doesn't exist or there is no saved state for a particular control. So the control ends up at -6dB and the volume is halved. I have reported this issue upstream here: http://thread.gmane.org/gmane.linux.alsa.devel/135504 but in the meantime I think the best course of action is to revert to the 3.12 situation and simply remove these controls. Therefore this pull request simply reverts commit 5be2fc20b101b5138c4f54a584dc11790ef16598.

popcornmix commented:
This is an upstream commit you are reverting. I'm not keen on doing that without confirmation it is correct.
We'll see what the view is from the upstream mailing list discussion. Let me know if a consensus is reached.

hmbedded committed:
ALSA: ControlName.txt: Add 'Analogue' as a prefix for volume controls

Currently 'Playback Volume' is the correct way to express an analogue
volume control. However, this control name has initialisation defaults
applied when using 'alsactl restore' and in some cases this is not
appropriate. An example would be a control that has a selection of
0db and -6dB of gain that is intended to set the fullscale ouput
voltage of a DAC. The TI pcm512x family of DAcs have such a control.
In this case the device/driver reset defaults are preferred.

Signed-off-by: Howard Mitchell <hm@hmbedded.co.uk>
Acked-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: Mark Brown <broonie@kernel.org>

hmbedded committed:
ASoC: pcm512x: Add 'Analogue' prefix to analogue volume controls

This is to ensure that 'alsactl restore' does not apply default
initialisation as the chip reset defaults are preferred.

Signed-off-by: Howard Mitchell <hm@hmbedded.co.uk>
Signed-off-by: Mark Brown <broonie@kernel.org>
Cc: stable@vger.kernel.org

hmbedded commented:
A consensus has been reached and as a result I submitted a couple of patches upstream: http://thread.gmane.org/gmane.linux.alsa.devel/135729. These have been accepted and have now made it to the linux-next tree. I'll update this PR with these patches shortly.

popcornmix commented:
Thanks.
